lib/Parser/Parse.cpp

```diff
@@ -1701,6 +1703,12 @@ void Parser::BindPidRefsInScope(IdentPtr pid, Symbol *sym, int blockId, uint max
             sym->SetIsModuleExportStorage(true);
         }
 
+        if (ref->GetFuncScopeId() != funcId && !sym->GetIsGlobal())
+        {
+            Assert(ref->GetFuncScopeId() > funcId);
```

Comment:
Yes, that's what it means. The parser is now tracking func ID's and scope ID's separately for each name reference. A func ID != the declaration's func ID indicates a reference in a nested function. (And since a function's ID is < all its child ID's, we can assert that the reference ID is > the declaration's ID.)